berkeley-mobile/Calendar/Campus/Cells/CampusEventCellView.swift

```diff
@@ -0,0 +1,139 @@
+        
+        cellImage.rightAnchor.constraint(equalTo: self.layoutMarginsGuide.rightAnchor).isActive = true
+        cellImage.topAnchor.constraint(equalTo: self.layoutMarginsGuide.topAnchor).isActive = true
+        cellImage.bottomAnchor.constraint(equalTo: self.layoutMarginsGuide.bottomAnchor).isActive = true
+        cellImage.widthAnchor.constraint(equalTo: self.widthAnchor, multiplier: 0.35).isActive = true
+        
+        leftVerticalStack.addArrangedSubview(nameLabel)
+        leftVerticalStack.addArrangedSubview(timeLabel)
+        // must constrain heights to prevent name label from taking up all the space
+        timeLabel.heightAnchor.constraint(equalToConstant: 14).isActive = true
+        locationLabel.heightAnchor.constraint(equalToConstant: 14).isActive = true
+    }
+    
+    /**
+     Update the values in the cell based on a new event (when collection/table is scrolled)
+     imageUpdate called to update cell image if cell is still visible
+     */
+    public func updateContents(event: EventCalendarEntry, cell: ImageViewCell) {
+        nameLabel.text = event.name
+        
+        // format date string to use "Today" or the actual date for other days
+        var dateString = ""
+        if event.date.dateOnly() == Date().dateOnly() {
+            dateString += "Today / "
+        } else {
+            let dateFormatter = DateFormatter()
+            dateFormatter.dateFormat = "MM/dd/yyyy"
+            dateString += dateFormatter.string(from: event.date) + " / "
+        }
+        let timeFormatter = DateFormatter()
+        timeFormatter.dateFormat = "h:mm a"
+        dateString += timeFormatter.string(from: event.date)
+        timeLabel.text = dateString
+        
+        if leftVerticalStack.arrangedSubviews.contains(locationLabel) {
+            if let location = event.location {
+                locationLabel.text = location
+            } else {
+                leftVerticalStack.removeArrangedSubview(locationLabel)
+            }
+        } else if let location = event.location {
+            leftVerticalStack.addArrangedSubview(locationLabel)
+            locationLabel.text = location
+        }
+        
+        cell.updateImage(item: event)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    let nameLabel: UILabel = {
+        let label = UILabel()
+        label.font = Font.bold(20)
+        label.textColor = Color.blackText
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.adjustsFontSizeToFitWidth = true
+        label.minimumScaleFactor = 0.6
+        label.numberOfLines = 0
+        return label
+    }()
+    
+    let leftVerticalStack: UIStackView = {
+        let stack = UIStackView()
+        stack.translatesAutoresizingMaskIntoConstraints = false
+        stack.axis = .vertical
+        stack.alignment = .leading
+        stack.distribution = .equalSpacing
```

Comment:
With equal spacing, when the name label doesn't take up as much space, the spacing between the items grow equally (see the spacing between the date and "Online" for the below two cells).
<img width="584" alt="Screen Shot 2020-10-11 at 11 14 44 PM" src="https://user-images.githubusercontent.com/41145903/95711743-49182680-0c18-11eb-92f6-8b146f191a6d.png">

Is this the behavior we want? Or would we want the date and online to both hug the bottom of the cell? Up to you.

Reply:
changed to bottom align date/location by moving them out of the stack